### PR TITLE
imx8mm: gpio driver update for imx8m-mini

### PR DESCRIPTION
**Pull request by neilsh-msft**
- add imx8m mini pin definitions to new header file
- store pullUp and pullDown PAD_CTL values in static class variables based on soc type instead of calculating from now non SOC-unique pad shift value
- add code to determine iMX8M Mini variant.  ANATOP_DIGPROG has changed location on iMX8M Mini.
- minor code beautification

### CODE REVIEW

File: driver/gpio/imxgpio/imxgpio.cpp [hunk 1]
```diff
@@ -1915,12 +1913,28 @@ NTSTATUS IMX_GPIO::configureCPUType (
         pinCount = IMX8M_GPIO_PIN_COUNT;
         pullShift = IMX8M_GPIO_PULL_SHIFT;
         pullMask = IMX8M_GPIO_PULL_MASK;
+        pullUp = IMX8M_GPIO_PULL_UP;
+        pullDown = (UINT32)-1;  // Force invalid value. iMX8M doesn't implement pull-down
```
Comment (hbeberman) [inline driver/gpio/imxgpio/imxgpio.cpp]: I think it'd be clearer to define an IMX_GPIO_PULL_INVALID in the IMX_GPIO_PULL enum, then explicitly list it above the default cases in PullModeToString and in GpioPullModeToImxPullMode.

**Review by hbeberman — CHANGES_REQUESTED** [review]

**Review by hbeberman — APPROVED** [review]

**Review by christopherco — APPROVED** [review]

File: driver/gpio/imxgpio/imxgpio.cpp (hunk 1 above)
Comment (neilsh-msft) [reply driver/gpio/imxgpio/imxgpio.cpp]: I've added a definition to the enumeration, and added it to PullModeToString, but it shouldn't be in GpioPullModeToImxPullMode.  IMX_GPIO_PULL reflects values we read or write to the IMX hardware + two flag values (DEFAULT and INVALID).  GpioPullModeToImxPullMode converts GPIO CLX value defs to IMX variants.

File: driver/gpio/imxgpio/imxgpio.cpp [hunk 2]
```diff
@@ -52,6 +52,7 @@ namespace {
                 return "PullDown";
             case IMX_GPIO_PULL_DISABLE:
                 return "PullNone";
+			case IMX_GPIO_PULL_INVALID:
```
Comment (hbeberman) [inline driver/gpio/imxgpio/imxgpio.cpp]: Should be 12 spaces instead of 3 tabs.

File: driver/gpio/imxgpio/imxgpio.cpp [hunk 3]
```diff
@@ -45,9 +46,13 @@ namespace {
             case IMX_GPIO_PULL_DEFAULT:
                 return "PullDefault";
             case IMX8M_GPIO_PULL_UP:
+            case IMX8MM_GPIO_PULL_UP:
                 return "PullUp";
+            case IMX8MM_GPIO_PULL_DOWN:
+                return "PullDown";
             case IMX_GPIO_PULL_DISABLE:
                 return "PullNone";
+            case IMX_GPIO_PULL_INVALID:
```
Comment (neilsh-msft) [inline driver/gpio/imxgpio/imxgpio.cpp:55]: IMX_GPIO_PULL_INVALID should only be found in the pullUp or pulldown static class members when pull up or pull down isn't supported by the hw.  It should not found natively in the HW, nor should it be generated by GpioPullModeToImxPullMode.  Hence the reason it still has an assert following it.

File: driver/gpio/imxgpio/imxgpio.cpp (hunk 2 above)
Comment (neilsh-msft) [reply driver/gpio/imxgpio/imxgpio.cpp]: Already fixed.

**Pull request merged**